src/Microsoft.Diagnostics.Monitoring.EventPipe/Configuration/MetricSourceConfiguration.cs

```diff
@@ -13,70 +13,85 @@
 
 namespace Microsoft.Diagnostics.Monitoring.EventPipe
 {
+    [Flags]
+    public enum MetricType
+    {
+        EventCounter = 0x1,
+        Meter = 0x2
+    }
+    
+    public sealed class MetricEventPipeProvider
+    {
+        public string Provider { get; set; }
+
+        public float IntervalSeconds { get; set; }
+
+        public MetricType Type { get; set; } = MetricType.EventCounter | MetricType.Meter;
+    }
+
     public sealed class MetricSourceConfiguration : MonitoringSourceConfiguration
     {
         private readonly IList<EventPipeProvider> _eventPipeProviders;
         public string SessionId { get; private set; }
 
+        public static readonly string[] DefaultProviders = new[] { SystemRuntimeEventSourceName, MicrosoftAspNetCoreHostingEventSourceName, GrpcAspNetCoreServer };
+
         public MetricSourceConfiguration(float metricIntervalSeconds, IEnumerable<string> customProviderNames)
+            : this(metricIntervalSeconds, customProviderNames?.Any() == true ? CreateProviders(metricIntervalSeconds, customProviderNames) :
+            CreateProviders(metricIntervalSeconds, DefaultProviders))
         {
-            RequestRundown = false;
-            if (customProviderNames == null)
-            {
+        }
+
+        public MetricSourceConfiguration(float metricIntervalSeconds, IEnumerable<MetricEventPipeProvider> customProviderNames, int maxHistograms = 20, int maxTimeSeries = 1000) {
```

Comment:
Do you care about having the flexibility to set different time intervals per-provider? This code path gets a bit confusing because we've got a top-level parameter but also every MetricEventPipeProvider is defining its own interval and then some of these input values appear to get ignored.

Could we simplify to MetricSourceConfiguration(float metricIntervalSeconds, IEnumerable<string> meterNames, IEnumerable<string> eventCounterProviderNames, ...)? If we did that we'd have one shared interval and perhaps no need for MetricEventPipeProvider class or MetricType enum?

Reply:
Was there a particular motivation for the per-provider intervals? I wasn't aware there was any demand for that kind of functionality so I am anticipating it would go unused even if we did add it.

MetricsEventSource doesn't support per-provider intervals so for the API to accurately reflect the underlying capabilities we'd end up with something asymmetric like MetricSourceConfiguration(float metricsIntervalSecs, IEnumerable<string> meterNames, IEnumerable<MetricEventPipeProvider> eventCounterProviders, ...). 

I don't want to hold you up over this. I am OK with the current code as-is or either of those alternative APIs. From what I know so far the option I like best is where we don't support per-provider intervals.

Reply:
The request on our side is tracked with https://github.com/dotnet/dotnet-monitor/issues/2870. My intent is to allow a hierarhical interval specification. I.e. you have a global interval, and then can change it for each provider. For meter-based specification, per provider configuration would be ignored. I will tweak this a bit to be clearer.

Reply:
https://github.com/dotnet/dotnet-monitor/issues/2870 - Is that request from customer feedback? The person who filed it is another VSDiag person so I can't tell who is being represented :) I might be totally wrong, but my understanding of the space so far is that the overwhelming majority of customers would not care about or use multiple intervals if you add support for it. Just for curiousity and learning sake I'd love to hear if you had feedback to the contrary, but as long as you are confident it will gain you some users go for it.